registry/handlers/context.go

```diff
@@ -33,6 +34,8 @@ type Context struct {
 	// handler *must not* start the response via http.ResponseWriter.
 	Errors errcode.Errors
 
+	lock sync.Mutex
+
```

Comment:
Don't extend this struct. It runs contrary to current best practices against contexts, and adding new fields makes it more difficult to refactor away in the future. Also, there is no need for a mutex here as a given instance of the struct is never shared among multiple concurrent requests.